zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/impl/DefaultContainer.kt

```diff
@@ -95,6 +95,17 @@ abstract class DefaultContainer(position: Position,
         return removalHappened
     }
 
+    override fun removeAllComponents(): Boolean {
+        val removalHappened = components.isNotEmpty()
+        components.clear()
```

Comment:
`Component`s have a [reference to their parent](https://github.com/Hexworks/zircon/blob/master/zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/InternalComponent.kt#L43) so you'd also need to clear that otherwise it would cause weird bugs.

I'd suggest adding this method to `Container`:

```kotlin
    fun removeAllComponents(): Boolean {
        val removalHappened = children.isNotEmpty()
        children.forEach { 
            removeComponent(it)
        }
        return removalHappened
    }
```